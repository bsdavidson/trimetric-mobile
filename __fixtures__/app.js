export const props = {
  dimensions: {
    screen: {
      width: 500,
      height: 600
    }
  },
  following: false,
  layerVisibility: {},
  loaded: true,
  routeShapes: [],
  selectedArrival: {},
  selectedArrivalVehicleInfo: {},
  selectedItem: {},
  selectedItemsInfo: [],
  selectedItemsViewHeight: 100,
  selectedVehicleInfo: {},
  stopPoints: [],
  vehiclePoints: []
};
