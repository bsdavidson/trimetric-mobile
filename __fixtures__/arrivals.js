export const selectedStop = {
  type: "stop",
  position: [-122.68931172788143, 45.51957800931885],
  item: {
    type: "Feature",
    properties: {stop_id: "5016", type: "stop"},
    geometry: {
      type: "Point",
      coordinates: [-122.68931172788143, 45.51957800931885]
    }
  }
};

export const selectedItem = {
  type: "stop",
  position: [-122.68931172788143, 45.51957800931885],
  item: {
    type: "Feature",
    properties: {stop_id: "5016", type: "stop"},
    geometry: {
      type: "Point",
      coordinates: [-122.68931172788143, 45.51957800931885]
    }
  }
};

export const arrivals = [
  {
    route_id: "51",
    route_short_name: "51",
    route_long_name: "Vista",
    route_type: 3,
    route_color: "",
    route_text_color: "",
    trip_id: "7908833",
    stop_id: "5016",
    headsign: "Portland",
    arrival_time: "16:22:00",
    departure_time: "16:22:00",
    vehicle_id: null,
    vehicle_label: null,
    vehicle_position: {
      lat: 0,
      lng: 0,
      bearing: 0,
      odometer: 0,
      speed: 0
    },
    date: "2018-01-19T00:00:00Z",
    trip_shape: {
      direction_id: 1,
      route_id: "51",
      color: "",
      points: [
        {lat: 45.493347, lng: -122.70053},
        {lat: 45.493347, lng: -122.700701},
        {lat: 45.516844, lng: -122.678826}
      ],
      trip_id: "7908833"
    }
  },
  {
    route_id: "51",
    route_short_name: "51",
    route_long_name: "Vista",
    route_type: 3,
    route_color: "",
    route_text_color: "",
    trip_id: "7908834",
    stop_id: "5016",
    headsign: "Portland",
    arrival_time: "16:52:00",
    departure_time: "16:52:00",
    vehicle_id: null,
    vehicle_label: null,
    vehicle_position: {
      lat: 0,
      lng: 0,
      bearing: 0,
      odometer: 0,
      speed: 0
    },
    date: "2018-01-19T00:00:00Z",
    trip_shape: {
      direction_id: 1,
      route_id: "51",
      color: "",
      points: [
        {lat: 45.490581, lng: -122.709817},
        {lat: 45.490589, lng: -122.710375},
        {lat: 45.516844, lng: -122.678826}
      ],
      trip_id: "7908834"
    }
  }
];

export const selectedItems = {
  type: "FeatureCollection",
  features: [
    {
      type: "Feature",
      properties: {stop_id: "5016", type: "stop"},
      geometry: {
        type: "Point",
        coordinates: [-122.68931172788143, 45.51957800931885]
      }
    }
  ]
};
