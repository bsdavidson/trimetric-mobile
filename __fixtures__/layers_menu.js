export const display = true;

export const bottom = 60;

export const selectedItemsViewHeight = 0;

export const layerVisibility = {
  routeShapes: true,
  buses: false,
  trains: true,
  stops: true,
  vehicleLabels: true,
  buildings: false
};

export const dimensions = {
  window: {fontScale: 1, width: 375, height: 667, scale: 2},
  screen: {fontScale: 1, width: 375, height: 667, scale: 2}
};
