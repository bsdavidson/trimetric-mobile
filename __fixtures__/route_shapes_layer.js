export const props = {
  visible: true,
  routeShapes: {
    type: "FeatureCollection",
    features: [
      {
        type: "Feature",
        properties: {color: "#CE0F69", route_id: "194"},
        geometry: {
          type: "MultiLineString",
          coordinates: [
            [
              [-122.680501, 45.53012],
              [-122.681354, 45.530079],
              [-122.681105, 45.530084]
            ]
          ]
        }
      },
      {
        type: "Feature",
        properties: {color: "#84BD00", route_id: "193"},
        geometry: {
          type: "MultiLineString",
          coordinates: [
            [
              [-122.671375, 45.493955],
              [-122.671107, 45.49395],
              [-122.698618, 45.529861],
              [-122.698646, 45.530509],
              [-122.698649, 45.530616]
            ],
            [
              [-122.685036, 45.514564],
              [-122.685257, 45.514157],
              [-122.685323, 45.514034],
              [-122.685339, 45.514005]
            ]
          ]
        }
      }
    ]
  },
  selectedArrival: null
};
