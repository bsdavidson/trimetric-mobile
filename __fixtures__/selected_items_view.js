// Selected Stop
export const stopData = [
  {
    type: "stop",
    stop: {
      id: "7749",
      code: "7749",
      name: "SW 6th & Broadway",
      desc: "Northbound stop in Portland (Stop ID 7749)",
      lat: 45.506859,
      lng: -122.684151,
      zone_id: "B",
      url: "http://trimet.org/#tracker/stop/7749",
      location_type: 0,
      parent_station: "",
      direction: "North",
      position: "Nearside",
      wheelchair_boarding: 0,
      distance: 0
    },
    arrivals: [
      {
        route_id: "8",
        route_short_name: "8",
        route_long_name: "Jackson Park/NE 15th",
        route_type: 3,
        route_color: "",
        route_text_color: "",
        trip_id: "7902197",
        stop_id: "7749",
        headsign: "NE Dekum",
        arrival_time: "14:12:07",
        departure_time: "14:12:07",
        vehicle_id: "2652",
        vehicle_label: "8  To NE Dekum",
        vehicle_position: {
          lat: 45.498703,
          lng: -122.68535,
          bearing: 224,
          odometer: 0,
          speed: 0
        },
        date: "2018-01-22T00:00:00Z",
        trip_shape: {
          direction_id: 1,
          route_id: "8",
          color: "",
          points: [
            {lat: 45.499398, lng: -122.685589},
            {lat: 45.571787, lng: -122.660853}
          ],
          trip_id: "7902197"
        },
        nextArrivals: [
          {
            route_id: "8",
            route_short_name: "8",
            route_long_name: "Jackson Park/NE 15th",
            route_type: 3,
            route_color: "",
            route_text_color: "",
            trip_id: "7902198",
            stop_id: "7749",
            headsign: "NE Dekum",
            arrival_time: "14:22:07",
            departure_time: "14:22:07",
            vehicle_id: "3142",
            vehicle_label: "8  To NE Dekum",
            vehicle_position: {
              lat: 45.497105,
              lng: -122.68317,
              bearing: 208,
              odometer: 0,
              speed: 0
            },
            date: "2018-01-22T00:00:00Z",
            trip_shape: {
              direction_id: 1,
              route_id: "8",
              color: "",
              points: [
                {lat: 45.496432, lng: -122.682258},
                {lat: 45.571787, lng: -122.660853}
              ],
              trip_id: "7902198"
            }
          },
          {
            route_id: "8",
            route_short_name: "8",
            route_long_name: "Jackson Park/NE 15th",
            route_type: 3,
            route_color: "",
            route_text_color: "",
            trip_id: "7902199",
            stop_id: "7749",
            headsign: "NE Dekum",
            arrival_time: "14:32:07",
            departure_time: "14:32:07",
            vehicle_id: null,
            vehicle_label: null,
            vehicle_position: {
              lat: 0,
              lng: 0,
              bearing: 0,
              odometer: 0,
              speed: 0
            },
            date: "2018-01-22T00:00:00Z",
            trip_shape: {
              direction_id: 1,
              route_id: "8",
              color: "",
              points: [
                {lat: 45.499398, lng: -122.685589},
                {lat: 45.571787, lng: -122.660853}
              ],
              trip_id: "7902199"
            }
          },
          {
            route_id: "8",
            route_short_name: "8",
            route_long_name: "Jackson Park/NE 15th",
            route_type: 3,
            route_color: "",
            route_text_color: "",
            trip_id: "7902200",
            stop_id: "7749",
            headsign: "NE Dekum",
            arrival_time: "14:42:44",
            departure_time: "14:42:44",
            vehicle_id: null,
            vehicle_label: null,
            vehicle_position: {
              lat: 0,
              lng: 0,
              bearing: 0,
              odometer: 0,
              speed: 0
            },
            date: "2018-01-22T00:00:00Z",
            trip_shape: {
              direction_id: 1,
              route_id: "8",
              color: "",
              points: [
                {lat: 45.499398, lng: -122.685589},
                {lat: 45.571787, lng: -122.660853}
              ],
              trip_id: "7902200"
            }
          }
        ]
      },
      {
        route_id: "8",
        route_short_name: "8",
        route_long_name: "Jackson Park/NE 15th",
        route_type: 3,
        route_color: "",
        route_text_color: "",
        trip_id: "7902198",
        stop_id: "7749",
        headsign: "NE Dekum",
        arrival_time: "14:22:07",
        departure_time: "14:22:07",
        vehicle_id: "3142",
        vehicle_label: "8  To NE Dekum",
        vehicle_position: {
          lat: 45.497105,
          lng: -122.68317,
          bearing: 208,
          odometer: 0,
          speed: 0
        },
        date: "2018-01-22T00:00:00Z",
        trip_shape: {
          direction_id: 1,
          route_id: "8",
          color: "",
          points: [
            {lat: 45.496432, lng: -122.682258},
            {lat: 45.571787, lng: -122.660853}
          ],
          trip_id: "7902198"
        }
      },
      {
        route_id: "8",
        route_short_name: "8",
        route_long_name: "Jackson Park/NE 15th",
        route_type: 3,
        route_color: "",
        route_text_color: "",
        trip_id: "7902199",
        stop_id: "7749",
        headsign: "NE Dekum",
        arrival_time: "14:32:07",
        departure_time: "14:32:07",
        vehicle_id: null,
        vehicle_label: null,
        vehicle_position: {
          lat: 0,
          lng: 0,
          bearing: 0,
          odometer: 0,
          speed: 0
        },
        date: "2018-01-22T00:00:00Z",
        trip_shape: {
          direction_id: 1,
          route_id: "8",
          color: "",
          points: [
            {lat: 45.499398, lng: -122.685589},
            {lat: 45.571787, lng: -122.660853}
          ],
          trip_id: "7902199"
        }
      },
      {
        route_id: "8",
        route_short_name: "8",
        route_long_name: "Jackson Park/NE 15th",
        route_type: 3,
        route_color: "",
        route_text_color: "",
        trip_id: "7902200",
        stop_id: "7749",
        headsign: "NE Dekum",
        arrival_time: "14:42:44",
        departure_time: "14:42:44",
        vehicle_id: null,
        vehicle_label: null,
        vehicle_position: {
          lat: 0,
          lng: 0,
          bearing: 0,
          odometer: 0,
          speed: 0
        },
        date: "2018-01-22T00:00:00Z",
        trip_shape: {
          direction_id: 1,
          route_id: "8",
          color: "",
          points: [
            {lat: 45.499398, lng: -122.685589},
            {lat: 45.571787, lng: -122.660853}
          ],
          trip_id: "7902200"
        }
      },
      {
        route_id: "8",
        route_short_name: "8",
        route_long_name: "Jackson Park/NE 15th",
        route_type: 3,
        route_color: "",
        route_text_color: "",
        trip_id: "7902201",
        stop_id: "7749",
        headsign: "NE Dekum",
        arrival_time: "14:52:44",
        departure_time: "14:52:44",
        vehicle_id: null,
        vehicle_label: null,
        vehicle_position: {
          lat: 0,
          lng: 0,
          bearing: 0,
          odometer: 0,
          speed: 0
        },
        date: "2018-01-22T00:00:00Z",
        trip_shape: {
          direction_id: 1,
          route_id: "8",
          color: "",
          points: [
            {lat: 45.496432, lng: -122.682258},
            {lat: 45.571787, lng: -122.660853}
          ],
          trip_id: "7902201"
        }
      },
      {
        route_id: "8",
        route_short_name: "8",
        route_long_name: "Jackson Park/NE 15th",
        route_type: 3,
        route_color: "",
        route_text_color: "",
        trip_id: "7902202",
        stop_id: "7749",
        headsign: "NE Dekum",
        arrival_time: "15:02:44",
        departure_time: "15:02:44",
        vehicle_id: null,
        vehicle_label: null,
        vehicle_position: {
          lat: 0,
          lng: 0,
          bearing: 0,
          odometer: 0,
          speed: 0
        },
        date: "2018-01-22T00:00:00Z",
        trip_shape: {
          direction_id: 1,
          route_id: "8",
          color: "",
          points: [
            {lat: 45.499398, lng: -122.685589},
            {lat: 45.571787, lng: -122.660853}
          ],
          trip_id: "7902202"
        }
      }
    ]
  }
];

export const following = true;
export const selectedItem = {
  type: "stop",
  position: [-122.68415115773678, 45.50685913695389],
  item: {
    type: "Feature",
    properties: {stop_id: "7749", type: "stop"},
    geometry: {
      type: "Point",
      coordinates: [-122.68415115773678, 45.50685913695389]
    }
  }
};

export const selectedItemIndex = 0;
export const selectedArrival = null;

export const selectedArrivalVehicleInfo = null;
export const selectedStopInfo = {
  id: "7749",
  code: "7749",
  name: "SW 6th & Broadway",
  desc: "Northbound stop in Portland (Stop ID 7749)",
  lat: 45.506859,
  lng: -122.684151,
  zone_id: "B",
  url: "http://trimet.org/#tracker/stop/7749",
  location_type: 0,
  parent_station: "",
  direction: "North",
  position: "Nearside",
  wheelchair_boarding: 0,
  distance: 0
};

// Selected Vehicle
export const vehicleData = [
  {
    type: "vehicle",
    vehicle: {
      trip: {trip_id: "7918798", route_id: "200"},
      vehicle: {id: "206", label: "Green Line to Clackamas"},
      position: {
        lat: 45.50877,
        lng: -122.68381,
        bearing: 331,
        odometer: 0,
        speed: 0
      },
      current_stop_sequence: 1,
      stop_id: "13140",
      current_status: 1,
      timestamp: 1516660571,
      congestion_level: 0,
      occupancy_status: 0,
      route_type: 0
    }
  }
];

export const selectedItemVehicle = {
  type: "vehicle",
  position: [-122.68381, 45.50877],
  item: {
    type: "Feature",
    properties: {
      bearing: 331,
      route_id: "200",
      icon: "tram",
      vehicle_id: "206",
      type: "vehicle",
      stop_id: "13140"
    },
    geometry: {type: "Point", coordinates: [-122.68381, 45.50877]}
  }
};

export const selectedVehicleInfo = {
  id: "13140",
  code: "13140",
  name: "Jackson St Turnaround",
  desc: "Westbound stop in Portland (Stop ID 13140)",
  lat: 45.508653,
  lng: -122.6836,
  zone_id: "R",
  url: "http://trimet.org/#tracker/stop/13140",
  location_type: 0,
  parent_station: "",
  direction: "West",
  position: "At",
  wheelchair_boarding: 0,
  distance: 0
};
