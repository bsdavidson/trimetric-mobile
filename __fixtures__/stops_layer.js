export const stopPoints = {
  type: "FeatureCollection",
  features: [
    {
      type: "Feature",
      properties: {type: "stop", stop_id: "13990"},
      geometry: {type: "Point", coordinates: [-122.398702, 45.552013]}
    },
    {
      type: "Feature",
      properties: {type: "stop", stop_id: "13991"},
      geometry: {type: "Point", coordinates: [-122.407683, 45.55363]}
    }
  ]
};

export const stops = [
  {
    id: "2",
    code: "2",
    name: "A Ave & Chandler",
    desc: "Eastbound stop in Lake Oswego (Stop ID 2)",
    lat: 45.420609,
    lng: -122.675671,
    zone_id: "B",
    url: "http://trimet.org/#tracker/stop/2",
    location_type: 0,
    parent_station: "",
    direction: "East",
    position: "Nearside",
    wheelchair_boarding: 0,
    distance: 0
  },
  {
    id: "landmark-2560",
    code: "",
    name: "SE Park Ave MAX Station",
    desc: "",
    lat: 45.430735,
    lng: -122.635065,
    zone_id: "",
    url: "",
    location_type: 1,
    parent_station: "",
    direction: "",
    position: "",
    wheelchair_boarding: 0,
    distance: 0
  }
];
