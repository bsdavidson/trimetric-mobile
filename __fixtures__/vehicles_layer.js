export const props = {
  filter: ["!=", "non_existing_attribute", "1"],
  vehiclePoints: {
    type: "FeatureCollection",
    features: [
      {
        type: "Feature",
        properties: {
          type: "vehicle",
          icon: "tram",
          vehicle_id: "101",
          route_id: "200",
          stop_id: "7601",
          bearing: 272
        },
        geometry: {type: "Point", coordinates: [-122.67175, 45.526817]}
      },
      {
        type: "Feature",
        properties: {
          type: "vehicle",
          icon: "tram",
          vehicle_id: "536",
          route_id: "200",
          stop_id: "8372",
          bearing: 298
        },
        geometry: {type: "Point", coordinates: [-122.579895, 45.534035]}
      },
      {
        type: "Feature",
        properties: {
          type: "vehicle",
          icon: "tram",
          vehicle_id: "537",
          route_id: "200",
          stop_id: "8347",
          bearing: 165
        },
        geometry: {type: "Point", coordinates: [-122.563545, 45.53022]}
      },
      {
        type: "Feature",
        properties: {
          type: "vehicle",
          icon: "tram",
          vehicle_id: "538",
          route_id: "90",
          stop_id: "10579",
          bearing: 145
        },
        geometry: {type: "Point", coordinates: [-122.59301, 45.58756]}
      }
    ]
  },
  labelsVisible: true
};

export const vehicles = [
  {
    trip: {trip_id: "7918797", route_id: "200"},
    vehicle: {id: "101", label: "Green Line to Clackamas"},
    position: {
      lat: 45.50887,
      lng: -122.68386,
      bearing: 347,
      odometer: 0,
      speed: 0
    },
    current_stop_sequence: 1,
    stop_id: "13140",
    current_status: 1,
    timestamp: 1516746492,
    congestion_level: 0,
    occupancy_status: 0,
    route_type: 0
  },

  {
    trip: {trip_id: "7919096", route_id: "200"},
    vehicle: {id: "537", label: "Green Line to City Ctr"},
    position: {
      lat: 45.435745,
      lng: -122.56782,
      bearing: 350,
      odometer: 0,
      speed: 0
    },
    current_stop_sequence: 1,
    stop_id: "13132",
    current_status: 1,
    timestamp: 1516746481,
    congestion_level: 0,
    occupancy_status: 0,
    route_type: 0
  },

  {
    trip: {trip_id: "7914971", route_id: "90"},
    vehicle: {id: "538", label: "Red Line to Beaverton"},
    position: {
      lat: 45.53185,
      lng: -122.56438,
      bearing: 292,
      odometer: 0,
      speed: 0
    },
    current_stop_sequence: 7,
    stop_id: "8371",
    current_status: 2,
    timestamp: 1516746477,
    congestion_level: 0,
    occupancy_status: 0,
    route_type: 0
  }
];
