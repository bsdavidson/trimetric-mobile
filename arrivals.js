import React, {Component} from "react";
import {connect} from "react-redux";
import {Text, View, StyleSheet} from "react-native";

function mapStateToProps(state) {
  return {
    arrivals: state.arrivals
  };
}

class Arrivals extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <View>
        <Text>RenderArrivals</Text>
      </View>
    );
  }
}

export default connect(mapStateToProps)(Arrivals);
