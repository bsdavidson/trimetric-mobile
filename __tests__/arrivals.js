import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import {Arrivals} from "../arrivals";
import {arrivals, selectedItem} from "../__fixtures__/arrivals";

jest.mock("../helpers");

it("renders arrivals component", () => {
  const nullArrivals = renderer.create(<Arrivals />).toJSON();
  expect(nullArrivals).toMatchSnapshot("null");

  const arr = renderer
    .create(<Arrivals arrivals={arrivals} selectedItem={selectedItem} />)
    .toJSON();
  expect(arr).toMatchSnapshot("arrivals");
});
