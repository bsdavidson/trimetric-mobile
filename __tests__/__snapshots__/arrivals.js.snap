// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrivals 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#CCCCCC",
        "borderRadius": 4,
        "marginBottom": 0,
        "padding": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Upcoming Arrivals
    </Text>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "arrival_time": "16:22:00",
          "date": "2018-01-19T00:00:00Z",
          "departure_time": "16:22:00",
          "headsign": "Portland",
          "nextArrivals": Array [
            Object {
              "arrival_time": "16:52:00",
              "date": "2018-01-19T00:00:00Z",
              "departure_time": "16:52:00",
              "headsign": "Portland",
              "route_color": "",
              "route_id": "51",
              "route_long_name": "Vista",
              "route_short_name": "51",
              "route_text_color": "",
              "route_type": 3,
              "stop_id": "5016",
              "trip_id": "7908834",
              "trip_shape": Object {
                "color": "",
                "direction_id": 1,
                "points": Array [
                  Object {
                    "lat": 45.490581,
                    "lng": -122.709817,
                  },
                  Object {
                    "lat": 45.490589,
                    "lng": -122.710375,
                  },
                  Object {
                    "lat": 45.516844,
                    "lng": -122.678826,
                  },
                ],
                "route_id": "51",
                "trip_id": "7908834",
              },
              "vehicle_id": null,
              "vehicle_label": null,
              "vehicle_position": Object {
                "bearing": 0,
                "lat": 0,
                "lng": 0,
                "odometer": 0,
                "speed": 0,
              },
            },
            undefined,
            undefined,
            0,
          ],
          "route_color": "",
          "route_id": "51",
          "route_long_name": "Vista",
          "route_short_name": "51",
          "route_text_color": "",
          "route_type": 3,
          "stop_id": "5016",
          "trip_id": "7908833",
          "trip_shape": Object {
            "color": "",
            "direction_id": 1,
            "points": Array [
              Object {
                "lat": 45.493347,
                "lng": -122.70053,
              },
              Object {
                "lat": 45.493347,
                "lng": -122.700701,
              },
              Object {
                "lat": 45.516844,
                "lng": -122.678826,
              },
            ],
            "route_id": "51",
            "trip_id": "7908833",
          },
          "vehicle_id": null,
          "vehicle_label": null,
          "vehicle_position": Object {
            "bearing": 0,
            "lat": 0,
            "lng": 0,
            "odometer": 0,
            "speed": 0,
          },
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "paddingTop": 5,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderBottomWidth": 0,
                  "borderColor": "#cccccc",
                  "borderWidth": 0,
                  "flex": 1,
                  "flexDirection": "row",
                  "marginBottom": 0,
                  "minHeight": 83,
                  "padding": 4,
                  "paddingLeft": 10,
                  "paddingRight": 0,
                },
                Object {
                  "borderColor": "rgba(204,204,204,0.3)",
                  "paddingLeft": 0,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 4,
                    "borderTopRightRadius": 0,
                    "justifyContent": "center",
                    "marginRight": 0,
                    "padding": 0,
                    "width": 45,
                  },
                  Object {
                    "backgroundColor": "rgba(204,204,204,0.3)",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontSize": 20,
                      "fontWeight": "bold",
                      "marginTop": 1,
                    },
                    Object {
                      "color": "rgba(50,50,50,1)",
                    },
                  ]
                }
              >
                51
              </Text>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../assets/bus.png",
                    }
                  }
                  style={
                    Object {
                      "height": 25,
                      "marginTop": 0,
                      "padding": 0,
                      "paddingRight": 10,
                      "resizeMode": "contain",
                      "width": 25,
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "alignSelf": "flex-start",
                  "flex": 1,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "padding": 5,
                    },
                    Object {
                      "backgroundColor": "rgba(204,204,204,0.3)",
                      "color": "rgba(50,50,50,1)",
                    },
                  ]
                }
              >
                Vista
                 to 
                Portland
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "marginBottom": 5,
                    "marginLeft": 5,
                    "marginTop": 5,
                  }
                }
              >
                Bus scheduled to arrive
                 
                in a minute
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 2,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "marginLeft": 4,
                    },
                    Object {
                      "borderColor": "rgba(204,204,204,0.3)",
                    },
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontSize": 11,
                        "padding": 2,
                        "paddingBottom": 0,
                        "width": 60,
                      },
                    ]
                  }
                >
                  future arrivals:
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "opacity": 0.5,
                        "padding": 2,
                      },
                      Object {},
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#222222",
                        "fontSize": 10,
                        "fontWeight": "bold",
                        "textAlign": "left",
                      }
                    }
                  >
                    in a minute
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#222222",
                        "fontSize": 10,
                        "fontWeight": "bold",
                        "textAlign": "left",
                      }
                    }
                  >
                    (scheduled)
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "opacity": 0.5,
                      "padding": 2,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#222222",
                        "fontSize": 10,
                        "fontWeight": "bold",
                        "textAlign": "left",
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`null 1`] = `null`;
